package.json

```diff
@@ -3,39 +3,50 @@
   "private": true,
   "type": "module",
   "scripts": {
-    "dev": "vite",
-    "build": "vite build",
-    "preview": "vite preview",
+    "start": "run-s build:* start:server",
+    "build:client": "vite build",
+    "build:server": "tsc --project tsconfig.server.json",
+    "start:server": "node dist/server/src/server/server.js",
+    "dev": "run-p dev:* start:devServer",
+    "dev:watchClient": "vite build --watch",
+    "dev:watchServer": "tsc --project tsconfig.server.json --watch",
```

Comment:
I also have run into an issue where changing a palette does not cause a re-build by `vite build --watch`. While it does trigger `nodemon` to restart, the new palette in `./pubic/palettes/some_palette.json` is not copied to `./dist/client/palettes/some_palette.json`.  The server needs to be manually stopped and restarted for the palette changes to be copied over to the `dist` folder.

I tried adding a `build.watch` property to `vite.config.ts`, like so, but it did nothing:
```
build: {
 watch: {
  include: "public/palettes/*.json"
  }
...
}
```

The [documentation for the `watch` property](https://vitejs.dev/guide/build#rebuild-on-files-changes) suggests looking at the [rollup watch configuration](https://rollupjs.org/configuration-options/#watch) for more options, which is where the idea to include `./public/palettes/*.json` came from.  However, that is entirely for files involved in rollup, and not static files.  So far, I have found no way to watch for changes in static files.  This [github issue](https://github.com/vitejs/vite/issues/4970) made a request for this feature, but the [conclusion](https://github.com/vitejs/vite/issues/4970#issuecomment-1500960824) was that it wasn't needed.

There is also this [stackoverflow](https://stackoverflow.com/questions/69626090/how-to-watch-public-directory-in-vite-project-for-hot-reload) article, but I have not had the time to do a deep dive into it.

Reply:
I also looked into this issue and followed the [stackoverflow solution](https://stackoverflow.com/questions/69626090/how-to-watch-public-directory-in-vite-project-for-hot-reload) you found to write a Vite plugin:

```
import { PluginOption } from "vite";

export default function deployPublicDir(): PluginOption {
  return {
    name: "public-to-dist",
    handleHotUpdate({ file, server }) {
      console.log("in handleHotUpdate: ", file);
      if (file.includes("palettes")) {
        server.ws.send({
          type: "custom",
          event: "public-file-updated",
        });
      }
    },
  };
};
```

This problem is the `handleHotUpdate()` option is never triggered when a file in `public` is updated. This seems because the `public` directory is never watched in the Vite watch mode:

* [Vite watch documentation](https://vitejs.dev/guide/build#rebuild-on-files-changes) says: `With the --watch flag enabled, changes to the vite.config.js, as well as any files to be bundled, will trigger a rebuild.` Changes to files to be bundled will trigger a rebuild, but files in `public` are not bundled via rollup. They are just copied to the destination root in the building process.
* The conclusion comment in the Vite [github issue](https://github.com/vitejs/vite/issues/4970) also says: `The conceptual model is that files in public are not watched for HMR during dev either.`

So, I'm not sure how the stackoverflow solution managed to get `handleHotUpdate()` triggered with a change to the `public` directory. I wonder if `i18n` package did something special in this case, or if the `public` directory in that example is a part of the Vite bundle?

I think for now we will need to manually restart the building process when there's a change to `public` directory.